patches/minecraft/net/minecraft/server/ReloadableServerResources.java.patch

```diff
@@ -1,5 +1,29 @@
 --- a/net/minecraft/server/ReloadableServerResources.java
 +++ b/net/minecraft/server/ReloadableServerResources.java
+@@ -28,18 +_,21 @@
+    private static final Logger f_206845_ = LogUtils.getLogger();
+    private static final CompletableFuture<Unit> f_206846_ = CompletableFuture.completedFuture(Unit.INSTANCE);
+    private final Commands f_206847_;
+-   private final RecipeManager f_206848_ = new RecipeManager();
++   private final RecipeManager f_206848_;
+    private final TagManager f_206849_;
+    private final PredicateManager f_206850_ = new PredicateManager();
+    private final LootTables f_206851_ = new LootTables(this.f_206850_);
+    private final ItemModifierManager f_206852_ = new ItemModifierManager(this.f_206850_, this.f_206851_);
+-   private final ServerAdvancementManager f_206853_ = new ServerAdvancementManager(this.f_206850_);
++   private final ServerAdvancementManager f_206853_;
+    private final ServerFunctionLibrary f_206854_;
+ 
+    public ReloadableServerResources(RegistryAccess.Frozen p_206857_, Commands.CommandSelection p_206858_, int p_206859_) {
+       this.f_206849_ = new TagManager(p_206857_);
+       this.f_206847_ = new Commands(p_206858_);
+       this.f_206854_ = new ServerFunctionLibrary(p_206859_, this.f_206847_.m_82094_());
++      // Forge: pass tagManager too
++      this.f_206848_ = new RecipeManager(this.f_206849_);
```

Comment:
Rather than passing in a tag manager, I would suggest passing in an IContext. It means we are constructing fewer instances, and if we want to add more to the context later we don't have to deprecate the old forge constructor. Have the default constructor set the field to an empty context.

Reply:
Implemented.